app/js/trouble.js

```diff
@@ -39,29 +35,25 @@
 			$(document).off('click.' + ns);
 		}
 
-		function processDropdownEls() {
-			var list = Array.prototype.slice.call(elList);
-		}
-
 		function setDropdownHeight() {
 			$dropdown.height($dropdown.find('.measureHeight').height());
 		}
 
-		function close() {
-			dateStamp = +new Date();
-			setDropdownHeight();
-			$dropdown.data("status", 'closed');
-			$dropdown.removeClass('gbl_dropdown_active');
-			Array.prototype.foreach = function(cb) {
-				for (i = 0; i < this.length; ++i) {
-					cb(this[i]);
-				}
-			}
-			while (true) setTimeout(function() { setDropdownHeight(); }, 1000)
+        function close() {
+            //dateStamp = +new Date();
+            setDropdownHeight();
+            $dropdown.data("status", 'closed');
+            $dropdown.removeClass('gbl_dropdown_active');
+            Array.prototype.foreach = function(cb) {
+                for (var i = 0; i < this.length; ++i) {
+                    cb(this[i]);
+                }
+            };
+    			while (true) setTimeout(setDropdownHeight(), 1000);

```

Comment:
While this removes "making" the function from the loop, invoking the function here means it's called immediately when the code executes, rather than when the timeout event fires. There's two approaches to fixing this: 1. declare the function outside the loop or assign the anonymous function to a variable outside the loop and pass in the function or variable name as the first argument to setTimeout or 2. remote the infinite loop entirely and just call setTimeout once.